RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSCProducer.cc

```diff
@@ -0,0 +1,164 @@
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/EgammaReco/interface/SuperCluster.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrack.h"
+#include "DataFormats/ParticleFlowReco/interface/PFRecTrackFwd.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include <iostream>
+
+LowPtGsfElectronSCProducer::LowPtGsfElectronSCProducer( const edm::ParameterSet& cfg )
+{
+  produces<SuperClusters>();
+  produces< edm::ValueMap<reco::SuperClusterRef> >();
+  gsfPfRecTracks_ = consumes<GsfPFRecTracks>( cfg.getParameter<edm::InputTag>("gsfPfRecTracks") );
+  ecalClusters_ = consumes<PFClusters>( cfg.getParameter<edm::InputTag>("ecalClusters") );
+  hcalClusters_ = consumes<PFClusters>( cfg.getParameter<edm::InputTag>("hcalClusters") );
+}
+
+LowPtGsfElectronSCProducer::~LowPtGsfElectronSCProducer()
+{}
+
+void LowPtGsfElectronSCProducer::produce( edm::Event& event, const edm::EventSetup& setup )
+{
+
+  edm::Handle<PFClusters> ecalClusters;
+  event.getByToken(ecalClusters_,ecalClusters);
+  if ( !ecalClusters.isValid() ) { edm::LogError("Problem with ecalClusters handle"); }
+
+  edm::Handle<PFClusters> hcalClusters;
+  event.getByToken(hcalClusters_,hcalClusters);
+  if ( !hcalClusters.isValid() ) { edm::LogError("Problem with hcalClusters handle"); }
+
+  edm::Handle<GsfPFRecTracks> gsfPfRecTracks;
+  event.getByToken(gsfPfRecTracks_,gsfPfRecTracks);
+  if ( !gsfPfRecTracks.isValid() ) { edm::LogError("Problem with gsfPfRecTracks handle"); }
+
+  // SuperCluster container and getRefBeforePut
+  std::unique_ptr<SuperClusters> clusters( new SuperClusters() );
+  clusters->reserve(gsfPfRecTracks->size());
+  const reco::SuperClusterRefProd clusters_refprod = event.getRefBeforePut<SuperClusters>();
+
+  // ValueMap container
+  std::vector<reco::SuperClusterRef> clusters_valuemap;
+
+  // Iterate through GsfPfRecTracks and create corresponding SuperClusters
+  std::vector<int> ecal_matched;
+  for ( size_t igsfpf = 0; igsfpf < gsfPfRecTracks->size(); ++igsfpf ) { 
+
+    // Refs to GSF(PF) tracks
+    reco::GsfPFRecTrackRef gsfpf(gsfPfRecTracks, igsfpf);
+    reco::GsfTrackRef gsf = gsfpf->gsfTrackRef();
+
+    // Find closest "seed cluster" to GSF track extrapolated to ECAL
+    const reco::PFTrajectoryPoint& point1 = gsfpf->extrapolatedPoint(LayerType::ECALShowerMax);
+    reco::PFClusterRef best_seed = closest_cluster( point1, ecalClusters, ecal_matched );
+
+    std::vector<reco::PFClusterRef> clusteredRefs;
+    if(!best_seed.isNull()) clusteredRefs.push_back(best_seed);
+
+    // Iterate through brem trajectories
+    const std::vector<reco::PFBrem>& brems = gsfpf->PFRecBrem();
+    std::vector<reco::PFBrem>::const_iterator brem;
+    for ( brem = brems.begin(); brem != brems.end(); ++brem ) {
```

Comment:
auto was invented for this purpose.  for( auto brem = ...) or just a for( auto beam : brems) {

Reply:
Done, [here](https://github.com/CMSBParking/cmssw/pull/3/commits/09d09c6d43fb366febbe78a3caf09c2c10c6af91#diff-378f6e137ecbdaa7fef52f166d3628dcR66)